### ISSUE TITLE
Add proper icon for sleeping heroes

### ISSUE THREAD

**Issue by ihhub** [ihhub/fheroes2#6554]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

The current icon is horrible and not intuitive (see top-right corner of hero's portrait):
![image](https://user-images.githubusercontent.com/19829520/216777626-9c0bc9eb-3cdc-4a76-91d0-28fde76521de.png)

This is out of style icon.

### Describe the possible solution

Add proper icon which would be understandable by any player that a hero is inactive / sleeping.

**Comment by Mr-Magonus** [comment]
I didn't even know sleeping heroes was possible in Heroes II.  What is the size & style that you would like?  If you don't have specifics, anything general.

**Comment by ihhub** [comment]
Hi @Branikolog , please provide art specification.

**Comment by Branikolog** [comment]
Hi, @ihhub and @Mr-Magonus !

First of all, I think we should consider that we have two different places where we need to indicate the "sleeping" status of a hero: the status window on the Adventure Map and the Hero Overview window.

In the Hero Overview window, the portrait is large enough to place a relatively big icon on top of it. However, in the Adventure Map status window, the hero portrait is much smaller. At the same time, the icon still needs to remain clearly visible, especially when playing at high resolutions or on smaller screens (for example, on mobile devices).

Because of this, I think we need to consider several possible approaches:

create two icons with significantly different sizes but a very similar visual design;
use some kind of visual effect applied directly to the hero portrait;
or combine these ideas by using an icon that partially or completely overlaps the portrait while still preserving the ability to recognize the hero itself.

If we choose the second or third approach, the effect or icon should look appropriate and remain recognizable on both portrait sizes.

@ihhub, could you please remind me if we previously discussed making hero portraits monochrome to indicate the "sleeping" status? Or are we still considering specifically adding an icon in the corner of the portrait?

**Comment by Mr-Magonus** [comment]
A pup tent with a Z coming out of it should do fine.  I can make an engraved icon in good & evil tones, small and large.  No need to reinvent the wheel for a sleeping icon.

**Comment by Branikolog** [comment]
@Mr-Magonus 

I agree that we probably shouldn't overcomplicate the visual indication of this mechanic. A simple and recognizable symbol is likely the best approach here.

The main challenge is making sure that the icon remains clearly readable in the adventure map status window while not covering too much of the small hero portrait. The same style icon could then also be used in the hero window, where there is much more space available for it.

**Comment by Mr-Magonus** [comment]
Just let me know the dimensions and I can crank it out in short order.

**Comment by Mr-Magonus** [comment]
@Branikolog I messaged you some images on Discord since I cannot attach items here.

**Comment by Mr-Magonus** [comment]
<img width="63" height="43" alt="Image" src="https://github.com/user-attachments/assets/2f89dfd4-f346-4586-8ee3-574fc85cd9d1" />
<img width="189" height="129" alt="Image" src="https://github.com/user-attachments/assets/3e031fa7-6f3d-4995-bc80-a7ae778631a7" />

Choose which one you like.  The smaller should overlap on the top right of the mana meter.  I can tweak things if need be.

If it's possible in the code, I recommend diagonal dithered grayscale LUT on the sleeping portraits.  That way the color can be preserved but it will be overall greyed out.

**Comment by Branikolog** [comment]
Hi, @Mr-Magonus 
Thanks for working on this element!

Overall, I really like the concept of the icon following the style of the buttons from the adventure map status window. It feels consistent with the existing UI language. 🔥 

Personally, I like the crescent moon variation more, but the second option is also very good and absolutely acceptable. However, in the full version with the tent, I have a small concern: the tent shape reminds me a bit of the letter "A". To avoid this association, perhaps we could try making the tent more vertically elongated, or add a small flag on top. The flag would also create a nice connection with the hero's flag on the adventure map, making it feel like the hero is resting in a tent with their own banner nearby. 
Also, I feel that the icon is slightly lacking in contrast. Perhaps we could try making the symbols inside a bit darker to improve readability.

<img width="144" height="74" alt="Image" src="https://github.com/user-attachments/assets/d37aa5ed-ec6f-4cce-9fdd-269d075a5b84" />

Overall, the larger versions look very good and are clearly visible in the hero window.

<img width="454" height="343" alt="Image" src="https://github.com/user-attachments/assets/0c936907-2803-4771-b93f-18defb7107e2" />

My main difficulty is with the small icon used in the hero list. Currently, it overlaps part of the mana meter, and personally I am not entirely comfortable with this placement. However, I also cannot think of a clearly better location.

The icon is 9x9 pixels, and when placed in the corner of the hero frame, it blends too much with the frame itself. In this case, I think the icon outline should be made much more contrasting, so these two graphical elements do not merge visually.

<img width="534" height="430" alt="Image" src="https://github.com/user-attachments/assets/6441ba25-7db8-4a56-86f4-6ec2b4395404" />

After trying different approaches, I found one option that I personally like the most: what if we reduce the icon size to 7x7 pixels (removing part of the outline) and place it completely inside the mana meter, without covering its black border? In this position, the icon feels like a separate UI element and remains quite readable (the lowest portrait on screenshot).

Regarding the partially covered mana meter: perhaps we could preserve some of its informational value by scaling the remaining visible part proportionally, so that it still represents the ratio between current and maximum mana.

What do you think, @ihhub and @zenseii ?

**Comment by zenseii** [comment]
@Branikolog and @Mr-Magonus. I think the moon is better as the letter Z isn't understood everywhere as sleeping. The easiest example is Japanese, and there are some Heroes 2 fans in Japan from my experience.

The background is the same as for buttons, so players might reasonably think that this can be pressed. It is only supposed to be a status icon. I liked @Branikolog's sketch for dawn because it makes a different border around the icon. To solve the mana bar problem he was describing I think we can make the icon a part of the frame

<img width="318" height="179" alt="Image" src="https://github.com/user-attachments/assets/146fa8f8-03a1-43e9-a08b-f211f432f6ae" />

Remember that there is an evil interface version of this too. So a gray/silver version would be necessary too.

The same goes for the larger hero portrait where. The outline and background color of the icon will have to be adjusted to the color of the frame around the portrait. In the Hero screen this would be the gold/yellow color of the frame. This is

**Comment by Branikolog** [comment]
@zenseii 
Instead of trying to make the icon visually separate from the frame, you made it intentionally become a part of the frame by moving it further into the corner. What an elegant solution!

It also avoids fully covering the mana bar, which solves the issue of preserving information about the hero's current mana. This feels like a very natural and well-integrated approach. Bravo, @zenseii!

I have been trying to find a good solution for this problem for quite some time, and I have to admit that I did not expect such an elegant approach.

I also agree with the golden version of the icon in the hero window. However, I still think the pictograms itself could be slightly darker to improve its readability and make the symbol stand out more clearly.

**Comment by zenseii** [comment]
Thanks for the praise, @Branikolog but it was really your sketch that gave me the idea. Also I agree about using darker brown color for the pictogram.

**Comment by Mr-Magonus** [comment]
> Remember that there is an evil interface version of this too. So a gray/silver version would be necessary too.

I have already submitted the gray/silver variation, if you scroll back up.

As an American, "Z" signifies sleep much more strongly than 🌙—on my mobile device, for an immediate example, Z is sleep mode and 🌙 is to darken the screen.  Since I have made both already, I might recommend using both for localization.

**Comment by Mr-Magonus** [comment]
@Branikolog 

Here is an updated version with increased contrast:

<img width="63" height="43" alt="Image" src="https://github.com/user-attachments/assets/cb628c21-99d9-4f45-ba46-fc9529e1eb4a" />

**Comment by Branikolog** [comment]
Thank you for the more contrast version. It already looks better and the icon is easier to read.
I don't think it would be worth creating different versions of the icon for different localizations. We already handle localization through text inside the engine, and having different graphical symbols for languages would be excessive. Also, the crescent moon concept works consistently for both the small and large versions, so players will immediately understand that they represent the same status. This is less obvious with the "Z" symbol and the "Z + tent" variation.

Regarding the final concept, I think the solution proposed by @zenseii — integrating the icon into the hero frame — is the most suitable approach.

For the adventure map, we would need a 7x7 icon in two color variations (good and evil). The top and right sides of the icon should closely match the hero frame border, so that this element looks like a natural continuation of the existing UI frame.
The left and bottom sides should have a black border line that integrates with the mana meter frame. This way, the icon will feel like it is embedded into the existing interface elements rather than simply placed on top of them.

The larger version in the hero window should also be revised using the golden colors matching the hero frame. The right and bottom sides, which touch the hero portrait, should have a border as well. This should make the element feel more like a hero property indicator rather than a clickable button.

By the way, I think this element in the hero window should have a right-click tooltip explaining the meaning of the status, so players can easily understand what the icon represents.

**Comment by zenseii** [comment]
I agree with all the points made by @Branikolog . I'd like to add that the background of the icon on the larger portrait on the hero screen should be yellow/golden like the frame, not the bright color it currently has.

**Comment by Mr-Magonus** [comment]
> I agree with all the points made by [@Branikolog](https://github.com/Branikolog) . I'd like to add that the background of the icon on the larger portrait on the hero screen should be yellow/golden like the frame, not the bright color it currently has.

I need clarification.  Can you highlight a sample of exactly the color you want matched?  Both assets I drew were sourced from existing UI buttons and are strictly palette correct as far as I know.

**Comment by zenseii** [comment]
@Mr-Magonus 
 
> I need clarification. Can you highlight a sample of exactly the color you want matched? Both assets I drew were sourced from existing UI buttons and are strictly palette correct as far as I know.

I've marked in red the colors that the backgrounds behind the icon need to be in:

<img width="543" height="190" alt="Image" src="https://github.com/user-attachments/assets/16f2e063-83a2-411c-a5c5-c9b41ab21eae" />

The evil interface one is different from the button background. I also suggest to make the background as uniform as possible without patterns like single pixels of a slightly darker or brighter shade.

The colors of the icon itself might need to be adjusted according to these new background colors. I wouldn't know so this is up to you.

**Comment by Mr-Magonus** [comment]
> I've marked in red the colors that the backgrounds behind the icon need to be in:

Okay, thanks.  I'll adjust the icons in the morning.

**Comment by Mr-Magonus** [comment]
@zenseii @Branikolog 

<img width="41" height="21" alt="Image" src="https://github.com/user-attachments/assets/f27d7b85-02fe-4c28-9c37-261c134c74cf" />

For viewing convenience:
<img width="170" height="90" alt="Image" src="https://github.com/user-attachments/assets/0e6826a4-dac2-4525-8d36-a80accd177b4" />

**Comment by Mr-Magonus** [comment]
The larger gold icon only needs 1 color as it's the same on evil & good.  Example:
<img width="380" height="252" alt="Image" src="https://github.com/user-attachments/assets/42fc687c-54c1-488a-822e-1e2754a2e035" />

And another option with increased gold contrast:
<img width="41" height="21" alt="Image" src="https://github.com/user-attachments/assets/ad37c2c6-047e-41f1-addf-6fd8d01d2757" />

**Comment by zenseii** [comment]
@Mr-Magonus the small icons with the black border look perfect now and are done.

<img width="65" height="36" alt="Image" src="https://github.com/user-attachments/assets/2b756bb1-dd94-447a-9942-cf163062052e" />

The golden one has perfect background and the moon is drawn well. The icon cannot have the button-like steps that I highlight in red below because the point of those steps on buttons is to make the buttons pop out of the background. This is not the goal of the sleep status icon. They need to be removed to avoid this effect where it now looks like the icon has a quasi-perspective depth effect where it looks like it is popping out of the portraits background but stopping at the golden frame of the portrait.


<img width="110" height="113" alt="Image" src="https://github.com/user-attachments/assets/8fa503b8-d959-4d71-99f6-b6df5796480c" />

We still need to have a border around this icon as is clearly evidenced by the quick edit I made below, and since we went for the outermost color of the frame, ie gold, it's harder to tell what the border of the icon needs to be.

<img width="293" height="272" alt="Image" src="https://github.com/user-attachments/assets/a82ebf12-1b6b-4bc3-977f-dd2e8734505c" />

By border I'm referring to the area highlighted in red here and which is black in the smaller icon:

<img width="92" height="91" alt="Image" src="https://github.com/user-attachments/assets/783a7590-e3b0-4962-8410-64992db58023" />

We could go for brown as that is used for border for the frame itself, but I'm not sure it makes sense here. We could make the icon brown instead and make the border gold but I'm still not sure this would look good. These are just ideas. I'm open to suggestions about how to solve this.

**Comment by zenseii** [comment]
Here's another edit showing an (unsuccessful) attempt of how to embed the icon into the frame 

<img width="326" height="297" alt="Image" src="https://github.com/user-attachments/assets/0fbbb217-8f1b-4960-96bc-1e2cf908b7e4" />

**Comment by Mr-Magonus** [comment]
Inverted to fit in that corner:
<img width="40" height="20" alt="Image" src="https://github.com/user-attachments/assets/f93f156a-4802-4538-8c45-18fc527f5faf" />

Example:
<img width="750" height="525" alt="Image" src="https://github.com/user-attachments/assets/f025c516-0d71-4b77-98e0-daa6880fa955" />

**Comment by zenseii** [comment]
@Mr-Magonus and @Branikolog. I think this looks perfect now. Good job @Mr-Magonus !

**Comment by Branikolog** [comment]
Hi, @zenseii and @Mr-Magonus 

Great job! Icon looks great!
